### ISSUE TITLE
Problems importing Variable Products

### ISSUE THREAD

**Issue by albertoalber** [woocommerce/woocommerce#18457]
## Prerequisites

- [x ] I have searched for similar issues in both open and closed tickets and cannot find a duplicate
- [x ] The issue still exists against the latest `master` branch of WooCommerce on Github (this is **not** the same version as on WordPress.org!)
- [x ] I have attempted to find the simplest possible steps to reproduce the issue
- [ ] I have included a failing test as a pull request (Optional)

## Steps to reproduce the issue

1. I export my WooCommerce Products List (they all are variable products).
2. I open it in Google Docs
3. I change the new prices, stock, whatever.
4. I save the CSV file.
5. I import the CSV file. Succesfully, as WooCommerce says.

## Expected/actual behavior

My Products are supposed to be changed. They have changed, indeed, but the only visible change is that variations have dissapeared. They are not set ("You have to set variations for the attributes", or similar").

## Isolating the problem

- [ ] This bug happens with only WooCommerce plugin active
- [ ] This bug happens with Flatsome
- [x ] I can reproduce this bug consistently using the steps above

## WordPress Environment

<details>
`
### WordPress Environment ###

Home URL: https://speedrill.com
Site URL: https://speedrill.com
WC Version: 3.2.6
Log Directory Writable: ✔
WP Version: 4.9.1
WP Multisite: –
WP Memory Limit: 512 MB
WP Debug Mode: –
WP Cron: ✔
Language: es_ES

### Server Environment ###

Server Info: LiteSpeed
PHP Version: 7.0.26
PHP Post Max Size: 8 MB
PHP Time Limit: 120
PHP Max Input Vars: 10000
cURL Version: 7.57.0
OpenSSL/1.0.2m

SUHOSIN Installed: –
MySQL Version: 5.6.38
Max Upload Size: 8 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 3.2.6
WC Database Prefix: wp_
MaxMind GeoIP Database: ✔
Tamaño total de la base de datos: 24.13MB
Tamaño de los datos de la base de datos: 16.28MB
Tamaño del índice de la base de datos: 7.85MB
wp_woocommerce_sessions: Datos: 0.00MB + Índice: 0.00MB
wp_woocommerce_api_keys: Datos: 0.00MB + Índice: 0.00MB
wp_woocommerce_attribute_taxonomies: Datos: 0.00MB + Índice: 0.00MB
wp_woocommerce_downloadable_product_permissions: Datos: 0.00MB + Índice: 0.00MB
wp_woocommerce_order_items: Datos: 0.00MB + Índice: 0.00MB
wp_woocommerce_order_itemmeta: Datos: 0.00MB + Índice: 0.01MB
wp_woocommerce_tax_rates: Datos: 0.00MB + Índice: 0.01MB
wp_woocommerce_tax_rate_locations: Datos: 0.00MB + Índice: 0.00MB
wp_woocommerce_shipping_zones: Datos: 0.00MB + Índice: 0.00MB
wp_woocommerce_shipping_zone_locations: Datos: 0.00MB + Índice: 0.01MB
wp_woocommerce_shipping_zone_methods: Datos: 0.00MB + Índice: 0.00MB
wp_woocommerce_payment_tokens: Datos: 0.00MB + Índice: 0.00MB
wp_woocommerce_payment_tokenmeta: Datos: 0.00MB + Índice: 0.00MB
wp_woocommerce_log: Datos: 0.00MB + Índice: 0.00MB
wp_commentmeta: Datos: 0.00MB + Índice: 0.00MB
wp_comments: Datos: 0.00MB + Índice: 0.01MB
wp_groups_capability: Datos: 0.01MB + Índice: 0.02MB
wp_groups_group: Datos: 0.00MB + Índice: 0.00MB
wp_groups_group_capability: Datos: 0.00MB + Índice: 0.00MB
wp_groups_user_capability: Datos: 0.00MB + Índice: 0.00MB
wp_groups_user_group: Datos: 0.00MB + Índice: 0.00MB
wp_icl_content_status: Datos: 0.00MB + Índice: 0.00MB
wp_icl_core_status: Datos: 0.00MB + Índice: 0.00MB
wp_icl_flags: Datos: 0.00MB + Índice: 0.00MB
wp_icl_languages: Datos: 0.00MB + Índice: 0.01MB
wp_icl_languages_translations: Datos: 0.10MB + Índice: 0.10MB
wp_icl_locale_map: Datos: 0.00MB + Índice: 0.00MB
wp_icl_message_status: Datos: 0.00MB + Índice: 0.00MB
wp_icl_mo_files_domains: Datos: 0.00MB + Índice: 0.00MB
wp_icl_node: Datos: 0.00MB + Índice: 0.00MB
wp_icl_reminders: Datos: 0.00MB + Índice: 0.00MB
wp_icl_strings: Datos: 2.29MB + Índice: 1.75MB
wp_icl_string_packages: Datos: 0.00MB + Índice: 0.00MB
wp_icl_string_pages: Datos: 0.35MB + Índice: 0.42MB
wp_icl_string_positions: Datos: 0.82MB + Índice: 0.17MB
wp_icl_string_status: Datos: 0.00MB + Índice: 0.00MB
wp_icl_string_translations: Datos: 0.00MB + Índice: 0.00MB
wp_icl_string_urls: Datos: 0.00MB + Índice: 0.01MB
wp_icl_translate: Datos: 1.31MB + Índice: 0.18MB
wp_icl_translate_job: Datos: 0.01MB + Índice: 0.01MB
wp_icl_translations: Datos: 0.19MB + Índice: 0.34MB
wp_icl_translation_batches: Datos: 0.00MB + Índice: 0.00MB
wp_icl_translation_status: Datos: 1.21MB + Índice: 0.02MB
wp_links: Datos: 0.00MB + Índice: 0.00MB
wp_m_communications: Datos: 0.00MB + Índice: 0.00MB
wp_m_coupons: Datos: 0.00MB + Índice: 0.00MB
wp_m_levelmeta: Datos: 0.00MB + Índice: 0.00MB
wp_m_membership_levels: Datos: 0.00MB + Índice: 0.00MB
wp_m_membership_news: Datos: 0.00MB + Índice: 0.00MB
wp_m_membership_relationships: Datos: 0.00MB + Índice: 0.00MB
wp_m_membership_rules: Datos: 0.00MB + Índice: 0.00MB
wp_m_member_payments: Datos: 0.00MB + Índice: 0.00MB
wp_m_pings: Datos: 0.00MB + Índice: 0.00MB
wp_m_ping_history: Datos: 0.00MB + Índice: 0.00MB
wp_m_subscriptionmeta: Datos: 0.00MB + Índice: 0.00MB
wp_m_subscriptions: Datos: 0.00MB + Índice: 0.00MB
wp_m_subscriptions_levels: Datos: 0.00MB + Índice: 0.00MB
wp_m_subscription_transaction: Datos: 0.00MB + Índice: 0.00MB
wp_m_urlgroups: Datos: 0.00MB + Índice: 0.00MB
wp_options: Datos: 1.40MB + Índice: 0.09MB
wp_postmeta: Datos: 6.00MB + Índice: 3.97MB
wp_posts: Datos: 2.20MB + Índice: 0.38MB
wp_redirection_404: Datos: 0.23MB + Índice: 0.10MB
wp_redirection_groups: Datos: 0.00MB + Índice: 0.00MB
wp_redirection_items: Datos: 0.02MB + Índice: 0.03MB
wp_redirection_logs: Datos: 0.00MB + Índice: 0.01MB
wp_sabadell_pago_referencia: Datos: 0.00MB + Índice: 0.00MB
wp_social_users: Datos: 0.00MB + Índice: 0.00MB
wp_termmeta: Datos: 0.03MB + Índice: 0.03MB
wp_terms: Datos: 0.01MB + Índice: 0.02MB
wp_term_relationships: Datos: 0.03MB + Índice: 0.07MB
wp_term_taxonomy: Datos: 0.01MB + Índice: 0.01MB
wp_usermeta: Datos: 0.02MB + Índice: 0.02MB
wp_users: Datos: 0.00MB + Índice: 0.01MB
wp_wcpdf_invoice_number: Datos: 0.00MB + Índice: 0.00MB
wp_yoast_seo_links: Datos: 0.00MB + Índice: 0.00MB
wp_yoast_seo_meta: Datos: 0.04MB + Índice: 0.04MB

### Post Type Counts ###

acf: 1
attachment: 534
blocks: 7
custom_css: 1
customize_changeset: 3
featured_item: 8
flamingo_contact: 19
flamingo_inbound: 64
ms_communication: 8
ms_event: 1
ms_membership: 2
nav_menu_item: 53
page: 34
post: 4
product: 300
product_variation: 3005
revision: 416
shop_coupon: 1
shop_order: 6
wc_membership_plan: 1
wpcf7_contact_form: 9

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (26) ###

WPML Multilingual CMS: por OnTheGoSystems – 3.9.0
Advanced Custom Fields: por Elliot Condon – 4.4.12
Akismet Anti-Spam: por Automattic – 4.0.2
TPV Virtual del Banco Sabadell: por Banco Sabadell – 2.0
Contact Form 7: por Takayuki Miyoshi – 4.9.2
Duplicate Post: por Enrico Battocchi – 3.2.1
Flamingo: por Takayuki Miyoshi – 1.7
Groups: por itthinx – 2.3.0
LiteSpeed Cache: por LiteSpeed Technologies – 1.7.2
Members: por Justin Tadlock – 2.0.2
Menu Icons: Icomoon: por Dzikri Aziz – 0.3.0
Menu Icons: por ThemeIsle – 0.11.0
Nextend Google Connect: por Roland Soos
Jamie Bainbridge – 1.6.1

Redirection: por John Godley – 2.10.1
Regenerate Thumbnails: por Alex Mills (Viper007Bond) – 3.0.1
WooCommerce - APG Weight and Postcode/State/Country Shipping: por Art Project Group – 2.2.2.2
WooCommerce Catalog Visibility Options: por Lucas Stark – 3.0.1 – 3.1.0 está disponible – No probado con la versión activa de WooCommerce
WooCommerce PayPal Express Checkout Gateway: por WooCommerce – 1.5.0
WooCommerce Multilingual: por OnTheGoSystems – 4.2.7.1
WooCommerce PDF Invoices & Packing Slips: por Ewout Fernhout – 2.1.3
WooCommerce: por Automattic – 3.2.6
WooSidebars: por WooThemes – 1.4.3
Yoast SEO: por Equipo Yoast – 6.1
WPML Media: por OnTheGoSystems – 2.2.2
WPML String Translation: por OnTheGoSystems – 2.7.0
WPML Translation Management: por OnTheGoSystems – 2.5.0

### Settings ###

API Enabled: ✔
Force SSL: –
Currency: EUR (€)
Currency Position: right
Thousand Separator: .
Decimal Separator: ,
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

### WC Pages ###

Base de la tienda: #638 - /tienda/
Carrito: #53 - /cart/
Finalizar compra: #52 - /checkout/
Mi cuenta: #51 - /my-account/
Términos y condiciones: #7688 - /terminos-condiciones/

### Theme ###

Name: Flatsome Child
Version: 3.2.5.1496750596
Author URL: http://www.uxthemes.com/
Child Theme: ✔
Parent Theme Name: Flatsome
Parent Theme Version: 3.4.2
Parent Theme Author URL: http://www.uxthemes.com/
WooCommerce Support: ✔

### Templates ###

Overrides: flatsome/woocommerce/archive-product.php
flatsome/woocommerce/cart/cart-empty.php
flatsome/woocommerce/cart/cart.php
flatsome/woocommerce/checkout/form-checkout.php
flatsome/woocommerce/checkout/form-coupon.php
flatsome/woocommerce/checkout/thankyou.php
flatsome/woocommerce/content-product.php
flatsome/woocommerce/content-product_cat.php
flatsome-child/woocommerce/content-single-product.php
flatsome/woocommerce/global/breadcrumb.php
flatsome/woocommerce/global/quantity-input.php
flatsome/woocommerce/global/wrapper-end.php
flatsome/woocommerce/global/wrapper-start.php
flatsome/woocommerce/loop/loop-end.php
flatsome/woocommerce/loop/loop-start.php
flatsome/woocommerce/loop/orderby.php
flatsome/woocommerce/loop/pagination.php
flatsome/woocommerce/loop/result-count.php
flatsome/woocommerce/loop/sale-flash.php
flatsome/woocommerce/myaccount/form-login.php
flatsome/woocommerce/myaccount/navigation.php
flatsome/woocommerce/notices/error.php
flatsome/woocommerce/notices/notice.php
flatsome/woocommerce/notices/success.php
flatsome/woocommerce/product-searchform.php
flatsome/woocommerce/single-product/photoswipe.php
flatsome/woocommerce/single-product/price.php
flatsome/woocommerce/single-product/product-image.php
flatsome/woocommerce/single-product/product-thumbnails.php
flatsome/woocommerce/single-product/related.php
flatsome/woocommerce/single-product/review.php
flatsome/woocommerce/single-product/sale-flash.php
flatsome/woocommerce/single-product/share.php
flatsome/woocommerce/single-product/short-description.php
flatsome/woocommerce/single-product/tabs/tabs.php
flatsome/woocommerce/single-product/title.php
flatsome/woocommerce/single-product/up-sells.php
flatsome/woocommerce/single-product-reviews.php
flatsome/woocommerce/single-product.php

`
</details>

**Comment by albertoalber** [comment]
![sin-titulo-3](https://user-images.githubusercontent.com/35367951/34874211-5b8ab09a-f798-11e7-8eb3-e7a87b402568.jpg)

**Comment by albertoalber** [comment]
Sorry, I think I closed this accidentally.

**Comment by mikejolley** [comment]
This is likely WP Multilingual - it duplicates data across languages. I am unsure to whether or not this supports our importer (you'd need to check).

If you do an import without that active you should see better results. Our default dummy data is importing without issue and includes variations. This also has unit tests so there is no issue with that part.

Also note, you'll want to update/merge when reimporting.

Thanks

**Comment by albertoalber** [comment]
Hello! Problem solved, now I would like to share.

First problem: export. I chose just the columns I needed (SKU, ID, Attributes, Price, Stock...). But now I choose everything and it works (basically, I think that is the column "Superior", don't know if it is in English, who makes that every attribute is in each product; before, every attribute became a product whrn import was done).
Second problem: when I edited database, I did it in Excel (I opened in Google Sheets, then export in .xls and included new data there). Now, I only use Google Sheets for everything, and no problem at all. 

IT WORKS!

Just one thing, related to WPML. I thought that editing products in our main language could be enough, that the products in English (or other language) would "take" the new data. I was wrong, each product in each language has its own register and must be modified separately.

Thanks.
